content/hardware/03.nano/boards/nano-esp32/tutorials/debugging/debugging.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Debugging with the Nano ESP32
+description: Get started with debugging using the Arduino Nano ESP32 with the IDE 2.
+author: Hannes Siebeneicher
+hardware:
+  - hardware/03.nano/boards/nano-esp32
+tags: [ESP32, Debugging, IDE]
+---
+
+Debugging is a crucial step especially when developing more complex projects. But even for beginners, it can be worth looking into as it gives you a lot more insights on what your code does at a specific time. 
+
+## What Is Debugging?
+
+Debugging is an essential skill for anyone working with technology, whether you're a software developer, a student learning to code, or someone who just wants to understand how computers work. Here's why it's so crucial:
+
+**Improving Performance**: Debugging helps identify and resolve issues that can slow down your software or make it behave unpredictably.
+
+**Ensuring Reliability**: Debugging ensures that your programs and devices work correctly and reliably, preventing crashes or malfunctions.
+
+**Saving Time and Frustration**: Fixing bugs early in the development process can save you hours, days, or even weeks of frustration down the road.
+
+**Learning Opportunity**: Debugging challenges your problem-solving skills and deepens your understanding of programming and technology.
+
+## Software & Hardware Needed
+
+- [Arduino Nano ESP32](https://store.arduino.cc/nano-esp32)
+- [Arduino IDE](/software/ide-v2)
+- [Arduino Nano ESP32 Core](https://github.com/arduino/arduino-esp32) (2.0.12 or newer)
+
+## IDE Setup
+
+In order to use the debugging feature on the Nano ESP32 you will need to download and install the IDE 2.2.0 or newer. After you install it, you have to configure the IDE as follows:
+
+-  **Tools** > **USB Mode** > **Debug mode (Hardware CDC)**
+
+![Debug Mode](./assets/debugMode.png)
+
+-  **Tools** > **Programmer** > **Esptool**
+
+![Select Programmer](./assets/programmer.png)
+
+-  **Sketch** > **Optimize for Debugging**
+
+![Optimize for Debugging](./assets/optimize.png)
+
+## Running a Debug Session
+
+If this is your first time debugging your code we recommend starting with the classic Blink example. It's a simple sketch but works great for understanding the basics of debugging.
+
+Before starting a debug session make sure to always do one of the following:
+
+- Either connect a jumper cable between the **GND** and the **B1** pins and press the reset button **once**. The RGB LED will turn on with a green or blue color. Inside **Tools**, the board will be shown as a random ESP32 board. 
+
+***This is because in this mode all ESP32 chips share the same identifier assigned to USB devices, therefore the IDE selects a random ESP32 board.***
```

Comment:
```suggestion
***This behavior demonstrated because as ESP32 chips share the same identifier assigned to USB devices, therefore the IDE selects a random ESP32 board.***
```

Reply:
Note that this is not the same meaning - they share the same VID/PID only when in debug mode; otherwise, in normal mode each ESP32 board reports a different VID/PID...